src/timepicker/timepicker.js

```diff
@@ -224,10 +237,9 @@ class TimePicker extends React.Component<TimePickerPropsT, TimePickerStateT> {
     // $FlowFixMe
     selectProps.overrides = selectOverrides;
 
-    const value =
-      this.props.value instanceof Date
-        ? this.buildSelectedOption(this.props.value, this.props.format)
-        : this.state.value;
+    const value = utils.isValid(this.props.value)
```

Comment:
This is the only change in here that doesn't map 1 to 1 to old date-fns api.  The problem is that now that the `value` could be any number of different objects, used a `instanceof` check here is going to be too strict.  `isValid` is the generic api `date.io` uses to achieve the same goal.

Reply:
makes sense 👍